gateway/api/views.py

```diff
@@ -314,14 +318,38 @@ def get_by_title(self, request, title):
         return Response(status=404)
 
     def _get_program_queryset_for_title_and_provider(
-        self, author, title: str, provider_name: Optional[str]
+        self,
+        author,
+        title: str,
+        provider_name: Optional[str],
+        type_filter: Optional[str],
     ):
         """Returns queryset for program for gived request, title and provider."""
         view_program_permission = Permission.objects.get(
             codename=VIEW_PROGRAM_PERMISSION
         )
 
         # Groups logic
+        if type_filter:
+            if type_filter == "catalog":
+                view_permission_criteria = Q(permissions=view_program_permission)
+                groups_with_view_permissions = Group.objects.filter(
+                    view_permission_criteria
+                )
+                groups_with_view_permissions_criteria = Q(
+                    instances__in=groups_with_view_permissions
+                )
+                provider_exists_criteria = ~Q(provider=None)
```

Comment:
Just for curiosity what it does the `~` in front of the `Q`?

Reply:
It's `NOT`.